src/main/java/cz/upce/fei/inptp/zz/entity/School.java

```diff
@@ -34,8 +32,10 @@ public boolean addStudentToCourseAction(Course course, Student student, TimeSlot
             return false;
         }
 
+        //check if specific timeSlot action is available in course
+        //check if course action has a free capacity for new student
         for (CourseAction action : course.getActions()) {
-            if (action.getTimeSlot().equals(timeSlot)) {
+            if (action.getTimeSlot().equals(timeSlot) && action.hasCapacity()) {
```

Comment:
Maybe better check will be nested if statement. Now if capacity condition is not satisfied it results to looping through rest of actions in course list.

Reply:
Solved.